docs/design/rebuild_out_synced_shards.md

```diff
@@ -0,0 +1,17 @@
+# Force rebuild out-synced Shards with broken Merkle Tree
+
+## Overview
+
+TODO
```

Comment:
Adding this will help understanding the code in this PR

Reply:
due to lack of time i will add it as a separate PR